packages/clerk-js/src/ui/signIn/SignInFactorTwo.tsx

```diff
@@ -54,9 +54,24 @@ function _SignInFactorTwo(): JSX.Element {
     if (status === 'verified' || !signIn) {
       return;
     }
+
+    const secondFactors = signIn.supportedSecondFactors
```

Comment:
Completely unrelated to this PR, but let's explore adding more helpers in `SignInResource` that'd handle common use cases like this, instead of letting the user filter and map manually.

This would also prevent the need for type assertions (eg`as PhoneCodeFactor`) - right now, types cannot be narrowed down as a result of a `.filter` call